codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -195,14 +195,26 @@ private FlowState connectionState() {
     }
 
     private static FlowState state(Http2Stream stream) {
-        checkNotNull(stream, "stream");
-        return stream.getProperty(FlowState.class);
+        if (stream instanceof FastPropertyAccess) {
```

Comment:
> Hmm it would be nice to not have to constantly do this when we know it will evaluate to true. instanceof may be fast but it still has to traverse the type hierarchy....and it isn't free.

`instanceof` actually is for free, the checkcast bytecode is the "expensive" thing. I haven't seen this method show up in profiling though, and quite possibly the JVM can optimize this cast away given that all stream implementations extend from this interface.